Release/include/cpprest/details/basic_types.h

```diff
@@ -43,20 +48,358 @@ typedef uint64_t size64_t;
+
+template<class _Ty, class _Dx = utility::secure_unique_ptr_deleter<_Ty>>
+using secure_unique_ptr = std::unique_ptr<_Ty, _Dx>;
+
+// TEMPLATE FUNCTION make_unique
+template<class _Ty, class _Dx = utility::unique_ptr_deleter<_Ty>, class... _Types> inline
+    typename std::enable_if<!std::is_array<_Ty>::value,
+    std::unique_ptr<_Ty, _Dx>>::type make_unique(_Types&&... _Args) {
+    // make a unique_ptr
+#if defined _WIN32 && !defined _DLL
+    _Ty* _Ptr = static_cast<_Ty*>(::HeapAlloc(::GetProcessHeap(), NULL, sizeof(_Ty)));
+    return std::unique_ptr<_Ty, _Dx>(new(_Ptr) _Ty(std::forward<_Types>(_Args)...));
+#else
+    return std::unique_ptr<_Ty, _Dx>(new _Ty(std::forward<_Types>(_Args)...));
+#endif
+}
+
+template<class _Ty, class _Dx = utility::unique_ptr_deleter<_Ty>> inline
+typename std::enable_if<std::is_array<_Ty>::value && std::extent<_Ty>::value == 0,
+    std::unique_ptr<_Ty, _Dx>>::type make_unique(size_t _Count) {
+    // make a unique_ptr
+#if defined _WIN32 && !defined _DLL
+    typedef typename std::remove_extent<_Ty>::type _Elem;
+
+    size_t _Size = _Count * sizeof(_Elem);
+    if (!std::is_trivially_destructible<_Elem>::value) {
+        // Extra space for new[] overhead for non-trivially destructible types
+        _Size += sizeof(size_t);
+    }
+
+    _Elem* _Ptr = static_cast<_Elem*>(::HeapAlloc(::GetProcessHeap(), NULL, _Size));
+    _Elem* _Array = new(_Ptr) _Elem[_Count]();
+
+    return std::unique_ptr<_Ty, _Dx>(_Array);
+#else
+    return std::unique_ptr<_Ty, _Dx>(new _Elem[_Count]());
+#endif
+}
+
+template<class _Ty, class _Dx = utility::unique_ptr_deleter<_Ty>, class... _Types>
+typename std::enable_if<std::extent<_Ty>::value != 0,
+    void>::type make_unique(_Types&&...) = delete;
+
+
+// TEMPLATE FUNCTION make_shared
+template<typename _Ty, typename _Alloc = utility::allocator<_Ty>, typename... _Types>
+inline std::shared_ptr<_Ty> make_shared(_Types&&... _Args) {
+    // make a shared_ptr
+    return std::allocate_shared<_Ty, _Alloc>(_Alloc(), std::forward<_Types>(_Args)...);
+}
+
+
+template<class _Ty, class _Alloc = utility::allocator<_Ty>>
+using vector = std::vector<_Ty, _Alloc>;
+
+template<class _Kty, class _Ty, class _Pr = std::less<_Kty>, class _Alloc = utility::allocator<std::pair<const _Kty, _Ty>>>
+using map = std::map<_Kty, _Ty, _Pr, _Alloc>;
+
+template<class _Kty, class _Ty, class _Hasher = std::hash<_Kty>,  class _Keyeq = std::equal_to<_Kty>, class _Alloc = utility::allocator<std::pair<const _Kty, _Ty>>>
+using unordered_map = std::unordered_map<_Kty, _Ty, _Hasher, _Keyeq, _Alloc>;
+
+template<class _Ty, class _Alloc = utility::allocator<_Ty>>
+using deque = std::deque<_Ty, _Alloc>;
+
+template<class _Ty, class _Container = utility::deque<_Ty>>
+using queue = std::queue<_Ty, _Container>;
+
+template<class _Ty, class _Traits = std::char_traits<_Ty>, class _Alloc = utility::allocator<_Ty>>
+using basic_string = std::basic_string<_Ty, _Traits, _Alloc>;
```

Comment:
Ah, I see this now; I don't think we can accept this change because it will be massively breaking to any existing customers.

Reply:
First of all thank you for the comments, I will try to go through them.

I don't consider this to be breaking existing customers compatibility because the utility:: should be equal to the std:: counterparts (i.e. utility::string = std::string) unless _WIN32 is defined and _DLL is not defined which is the case for /MT, which none of the existing customers would be using anyway, please correct me if i am wrong.